projects/rocprim/test/rocprim/test_utils_assertions.hpp

```diff
@@ -48,6 +48,56 @@
 
 namespace test_utils {
 
+#if ROCPRIM_HAS_INT128_SUPPORT
+template<class T>
+using is_int128 = std::is_same<rocprim::int128_t, typename std::remove_cv<T>::type>;
+template<class T>
+using is_uint128 = std::is_same<rocprim::uint128_t, typename std::remove_cv<T>::type>;
+#else
+template<class T>
+using is_int128 = std::false_type;
+template<class T>
+using is_uint128 = std::false_type;
+#endif // ROCPRIM_HAS_INT128_SUPPORT
+    
+namespace {
+// On Windows, GTest doesn't provide the appropriate overloads
+// for printing 128 bit types. As a result, there may be linker errors
+// if you use ASSERT_EQ with these types. In some situations, we may also
+// want to avoid printing other types on Windows.
+// The function below will avoid ASSERT_EQ when use_assert_true is set to true.
+// It does this by performing the comparison beforehand, storing it in a bool,
+// and then passing that to ASSERT_TRUE. This way, if the values are not equal,
+// only the bool will be printed.
+
+// These are for convenience when constructing a bool at runtime to pass in for use_assert_true.
+#if defined(_WIN32)
+constexpr bool is_win32 = true;
+#else
+constexpr bool is_win32 = false;
+#endif
+
+template <class T>
+void inline protected_assert_eq(T val, T expected, const bool use_assert_true, size_t index=-1)
```

Comment:
If we move `const bool use_assert_true` into the template we can guarantee that it won't emit the extra `use_assert_true` condition. The compiler will *probably* optimize it without it though. So feel free to reject this suggestion.

```suggestion
template <typename T>
constexpr bool is_printable = 
    test_utils::is_int128<T>::value || 
    test_utils::is_uint128<T>::value ||
    // could be moved out as 'using is_double_custom_type = ...'
    std::is_same_v<typename std::remove_cv<T>::type, common::custom_type<double,double,1>>;

template <class T, bool UseGTestAssert = is_printable<T>>
void inline protected_assert_eq(T val, T expected, const bool use_assert_true, size_t index=-1)
```

We can then also use `if constexpr (UseGTestAssert)`.

Reply:
Thanks Nara, I've pushed another commit with your suggested changes.
We do need this fix in 7.0, since it's a blocker for us, so it might be nice to just merge this PR since it's ready to go. If you're ok with merging your refactoring to develop after the mergeback, that would be great - feel free to wipe out this change there in place of yours if you like.